### PR TITLE
parser: support `ALTER TABLE ALTER CHECK`, `ALTER TABLE DROP CHECK` syntax

### PR DESCRIPTION
**Pull request by spongedu**
### What problem does this PR solve? 

for #497 , suport `ALTER TABLE ALTER CHECK`, `ALTER TABLE DROP CHECK` syntax.

### Check List 

Tests 

 - Unit test
 - Integration test
 - Manual test (add detailed scripts or steps below)

Code changes

N / A

Side effects

N / A

Related changes

N / A

### CODE REVIEW

**Comment by leoppro** [comment]
LGTM

**Comment by leoppro** [comment]
@tangenta PTAL

File: parser.y [hunk 1]
```diff
@@ -1595,6 +1595,36 @@ AlterTableSpec:
 		yylex.AppendError(yylex.Errorf("The WITH/WITHOUT VALIDATION clause is parsed but ignored by all storage engines."))
 		parser.lastErrorAsWarn()
 	}
+| "ALTER" "CHECK" Identifier EnforcedOrNot
```
Comment (tangenta) [inline parser.y]: These indentions are inconsistent here. Could you fix it?

Comment (spongedu) [reply parser.y]: > These indentions are inconsistent here. Could you fix it?

ok

File: parser.y [hunk 2]
```diff
@@ -1607,6 +1607,33 @@ AlterTableSpec:
...
```
Comment (tangenta) [inline parser.y]:
```suggestion
|	"ALTER" "CHECK" Identifier EnforcedOrNot
```
Sorry for my unclear suggestion. I mean here should be a tab, as most of the other productions.

File: parser_test.go [hunk 3]
```diff
@@ -2284,6 +2284,13 @@ func (s *testParserSuite) TestDDL(c *C) {
 		{"ALTER TABLE t RENAME KEY a TO b;", true, "ALTER TABLE `t` RENAME INDEX `a` TO `b`"},
 		{"ALTER TABLE t RENAME INDEX a TO b;", true, "ALTER TABLE `t` RENAME INDEX `a` TO `b`"},
 
+		// For #497
+
```
Comment (tangenta) [inline parser_test.go]: Please remove this blank line.

Comment (spongedu) [reply parser_test.go]: > Please remove this blank line.

OK.

**Pull request merged**